### PR TITLE
Add Multi-ASIC build options

### PR DESCRIPTION
**Pull request by theasianpianist**
**- What I did**
Add an option to build a multi-ASIC SONIC image
**- How I did it**
Add a multi-ASIC variable into the proper Makefiles
Create separate folders for single and multi-instance systemd services
**- How to verify it**
Call `make all MULTI_ASIC=y` should yield a multi-ASIC build
**- Description for the changelog**

Adds multi-ASIC build option

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: slave.mk [hunk 1]
```diff
@@ -668,6 +714,7 @@ $(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : \
 		rm -f $($(docker:-dbg.gz=.gz)_CONTAINER_NAME).sh
 		rm -f $($(docker:-dbg.gz=.gz)_CONTAINER_NAME).service
 	)
+	rm -f updategraph.service
```
Comment (pavel-shirshov) [inline slave.mk]: Why?

File: files/build_templates/docker_image_ctl.j2 [hunk 2]
```diff
@@ -198,8 +198,10 @@ start() {
         --tmpfs /tmp \
 {%- endif %}
         --tmpfs /var/tmp \
+{%- if multi_instance == "False" %}
         --hostname "$HOSTNAME" \
-        --name={{docker_container_name}} {{docker_image_name}}:latest || {
+{%- endif %}
```
Comment (pavel-shirshov) [inline files/build_templates/docker_image_ctl.j2]: so --hostname is still not compatible with --net?

File: slave.mk [hunk 3]
```diff
@@ -630,15 +635,51 @@ $(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : \
...
+		else
+			export multi_asic="False"
+		fi
+
 		if [ -f files/build_templates/$($(docker:-dbg.gz=.gz)_CONTAINER_NAME).service.j2 ]; then
 			j2 files/build_templates/$($(docker:-dbg.gz=.gz)_CONTAINER_NAME).service.j2 > $($(docker:-dbg.gz=.gz)_CONTAINER_NAME).service
 		fi
+
+		if [ "$(MULTI_ASIC)" = "y" ]; then
+			export multi_instance="True"
```
Comment (pavel-shirshov) [inline slave.mk]: what is the difference between
multi_asic vs multi_instance

File: slave.mk (hunk 1 above)
Comment (theasianpianist) [reply slave.mk]: `updategraph.service` wasn't getting deleted after the build finished

File: slave.mk (hunk 3 above)
Comment (theasianpianist) [reply slave.mk]: multi_asic indicates that we are building an image for a device with multiple ASICs. Multi_instance means we need multiple copies of a particular container

File: Makefile.work [hunk 4]
```diff
@@ -128,6 +128,7 @@ SONIC_BUILD_INSTRUCTION :=  make \
                            HTTPS_PROXY=$(https_proxy) \
                            SONIC_ENABLE_SYSTEM_TELEMETRY=$(ENABLE_SYSTEM_TELEMETRY) \
                            EXTRA_JESSIE_TARGETS=$(EXTRA_JESSIE_TARGETS) \
+                           MULTI_ASIC=$(MULTI_ASIC) \
```
Comment (lguohan) [inline Makefile.work]: I am not expecting to have this as a build option. 

the image should be able to adapt in runtime to figure out whether the platform is multi-asic or single asic and adapt to it.

**Pull request closed without merging**